src/cdk/a11y/focus-trap/focus-trap.ts

```diff
@@ -128,8 +137,30 @@ export class FocusTrap {
   }
 
   /**
-   * Waits for the zone to stabilize, then either focuses the first element that the
-   * user specified, or the first tabbable element.
+   * Waits for the zone to stabilize, then focuses the first element that matches the
+   * CSS selector.
+   * @returns Returns a promise that resolves with a boolean, depending
+   * on whether focus was moved successfully.
+   */
+  focusByCssSelectorWhenReady(selector: string, options?: FocusOptions): Promise<boolean> {
```

Comment:
Similar to the enum, I don't think this method, `focusFirstHeader`, `forceFocus`, etc. belong in the focus trap. The `focusFirstTabbableElement` methods make sense here because, when trapping, focus always wraps around to the first/last tabbable elements in the dialog, regardless of where the initial focus lands. But outside of that, the trap doesn't really care where the initial focus ends up. 

Also similarly, I think both of these methods belong in the nonexistent cdk/dialog. So, I would again duplicate them in each component and use this as another signal that we need to get cdk/dialog created.